### ISSUE TITLE
libjxl-0.6.1 fails to build on i686 architechture in nixos-unstable

### ISSUE THREAD

**Issue by llirium** [NixOS/nixpkgs#154832]
### Describe the bug
libjxl-0.6.1 fails to build on i686-based ASUS Eee PC 1000HE

### Steps To Reproduce
1. enable lxqt and lightdm in `configuration.nix`
2. `nixos-rebuild switch`

### Expected behavior
compilation succeeds.

### Additional context
Build log was enormous, so the log has been trimmed to show only Failed tests:
[libjxl-0.6.1_build_errors_trimmed.log](https://github.com/NixOS/nixpkgs/files/7859074/libjxl-0.6.1_build_errors_trimmed.log)

### Notify maintainers
@nh2

### Metadata
 - system: `"i686-linux"`
 - host os: `Linux 5.10.90, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.5.1`
 - channels(root): `"nixos-22.05pre344851.0ecf7d41481"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by trofi** [comment]
Might be a side-effect of excessive precision on i387 FPU: https://gcc.gnu.org/wiki/FAQ#PR323

The following hack allows me to build `pkgsi686Linux.libjxl` successfully:

```diff
--- a/pkgs/development/libraries/libjxl/default.nix
+++ b/pkgs/development/libraries/libjxl/default.nix
@@ -49,6 +49,8 @@ stdenv.mkDerivation rec {
     })
   ];

+  NIX_CFLAGS_COMPILE = [ "-msse" "-mfpmath=sse" ];
+
   nativeBuildInputs = [
     asciidoc # for docs
     cmake
```

**Comment by NickSeagull** [comment]
How can one add those flags without forking Nixpkgs?

I'm trying to enable xserver, but it's failing due to this issue (libjxl not building)

**Comment by trofi** [comment]
One of the options would be to add an override via overlay: https://nixos.wiki/wiki/Overlays

Something like (untested):

```nix
{ config, pkgs, lib, ... }:
{
   nixpkgs.overlays = [ (final: prev:
     libjxl = libjxl.overrideAttrs (oa: { NIX_CFLAGS_COMPILE = "-msse -mfpmath=sse"; })
   ) ];
}
```

**Comment by sg2002** [comment]
This has been fixed ever since 2363be9.
